### PR TITLE
refactor typings for readability

### PR DESCRIPTION
**Pull request by phryneas**
This should make the typings more readable for someone just diving into this lib's typings, as we already discussed on twitter :)

### CODE REVIEW

File: src/createAction.ts [hunk 1]
```diff
@@ -6,51 +6,55 @@ import { Action } from 'redux'
...
+export type ActionCreatorWithoutPayload<T extends string = string> = WithTypeProperty<T, () => PayloadAction<undefined, T>>;
+
+export type ActionCreatorWithPayload<P, T extends string = string> =
+  WithTypeProperty<T, <PT extends P>(payload: PT) => PayloadAction<PT, T>>;
+
 /**
  * An action creator that produces actions with a `payload` attribute.
  */
 export type PayloadActionCreator<
   P = any,
```
Comment (phryneas) [inline src/createAction.ts]: I could go and rename P to payload, T to Type and PA to PrepareActionFn here, but I'm not really sure if it makes things better or worse in the usages of these below.
Do you have an opinion there?

**Comment by phryneas** [comment]
@markerikson what's your status on this? From my side this would be ready if you don't have any additional comments.

**Comment by phryneas** [comment]
Okay :) Just wanted to make sure this didn't get buried ^^

**Comment by markerikson** [comment]
Sure, let's do this.

**Pull request merged**